### DISCUSSION TITLE
Password protect a page that is shared to the internet

### DISCUSSION THREAD

**Discussion by RickCogley** [outline/outline#1781] [Ideas]
Currently, the internet sharing scheme is a typical one, relying on "security by obscurity" by setting the URL as a UUID. These days, many systems offer this functionality. 

Problem is, search engines don't always honor "no index" metadata (if even Outline is setting that for so-shared pages), and people can share the URIs of such pages via various means. When the link goes into chat or email, voilà, the link is now in a backup somewhere. 

It would be good if Outline could allow users to set an arbitrary password to a shared page. When a person clicks on such a URL, Outline would prompt for a password, a la `http-access`. 

Of course, guest access is one way to handle this, but doing it with a password means, you can allow a bunch of people to see it for a limited period, and easily block access simply by changing the password.

**Comment by tommoor** [comment]
You can revoke and regenerate share links currently, so I think the "limited period" part is already covered by that. I can understand wanting the option for a password though and I don't see any harm in us including the functionality.

It seems like something that might be good for a community member to work on and we can guide through implementation.
